src/devana/utility/init_params.py

```diff
@@ -0,0 +1,44 @@
+from typing import Callable, Any, Optional, Set
+from inspect import signature, BoundArguments
+from functools import wraps
+
+
+def init_params(skip: Optional[Set[str]] = None) -> Callable:
+    """A decorator that assigns method parameters as instance attributes.
+    The attribute must have a setter or exist as an instance or class variable."""
+
+    def decorator(_classmethod: Callable) -> Callable:
+
+        def has_setter(instance: object, name: str) -> bool:
+            """Checks if the class attribute is a property with a defined setter method."""
+            try:
+                maybe_property = getattr(instance.__class__, name)
+            except AttributeError:
```

Comment:
Maybe it's easier to check if the attribute exists? Unless you're protecting yourself from other errors here?

Reply:
This makes no sense. `hasattr` calls the identical code underneath and returns `True` if it does not raise an exception.
We can use this trick, BUT use the object, if it exists, without an additional call to `geattr`.

Reply:
Checking things with exceptions is generally nasty. It's half the trouble when it wraps a function, hence my comment.